docs/fhir-web-docker-deployment.md

```diff
@@ -0,0 +1,210 @@
+# FHIR WEB DOCKER DEDPLOYMENT
+
+## Prerequisites
+
+---
+
+- A well configured [keycloak server](https://hub.docker.com/r/onaio/keycloak)
+- A well configured [Hapi FHIR server](https://github.com/opensrp/hapi-fhir-jpaserver-starter)
+- A reverse proxy server port forwarding `port 3000`
+
+## Background
+
+---
+
+- Whenever we cut a [FHIR web release](https://github.com/opensrp/web/releases) we use [github actions](/.github/workflows/docker-publish.yml) to build, tag, and push a docker image based on our [Dockerfile](/Dockerfile) specification to [docker hub](https://hub.docker.com/r/opensrp/web/tags)
+
+- This means that all our releases have a corresponding docker image on docker hub ready for use. E.g [release v2.3.4](https://github.com/opensrp/web/releases/tag/v2.3.4) has the corresponding [docker image tag v2.3.4](https://hub.docker.com/layers/opensrp/web/v2.3.4/images/sha256-de13e5482194d76fd22980e73cda0e4f77d6b59aab868130dea24c3b609aa312?context=explore)
+
+- These images are self contained, composed of the compiled [express authentication server](https://github.com/onaio/express-server) and the compiled [React FHIR Web front-end](https://github.com/opensrp/web) codebase.
+
+## Configuration
+
+---
+
+- To deploy FHIR Web you need to run a FHIR web docker image as a docker container and pass configuration environment variables to it.
+
+- Based on FHIR web architecture, the express server handles authentication (to Keycloak) and only serves the react front-end if the user is properly authenticated
+
+- The express server is configured as a first class docker service, by passing environment variables to the container.
+
+- The react front-end receives a config file, [config.js](https://github.com/opensrp/web/blob/master/app/public/config.js), with environment variables that are read into the browser's windows environment at run time. The config file is created and injected by [confd utility](https://github.com/kelseyhightower/confd) reading a docker volume file (`config.js.tpl`) holding the react app configurations.
+
+- E.g. use the following docker command paired with a volume config file
+
+  ```docker
+  docker run \
+  --volume $(pwd)/config.js.tpl:/etc/confd/templates/config.js.tmpl \
+  --publish 3000:3000 \
+  --env EXPRESS_OPENSRP_CLIENT_ID=example-keycloak-client-id \
+  --env EXPRESS_OPENSRP_CLIENT_SECRET=example-keycloak-client-secret \
+  `#...rest of the express server envs` \
+  --name fhir-web opensrp/web:v2.3.2
+  ```
+
+  - With the volume file in the same directory, as below:
+
+  ```js
+  // config.js.tpl
+
+  window._env_ = {
+    REACT_APP_WEBSITE_NAME: 'The best FHIR Web Deployment Ever!',
+    REACT_APP_OPENSRP_WEB_VERSION: 'v2.3.2',
+    // rest of the react front-end envs
+  };
+  ```
+
+- An even easier configuration method is to use a docker-compose file with a volume file
+
```

Comment:
Can we make the 2 ways of running the container more distinctive for example using the document hierarchy, structure or styling; e.g. using headers.